lbrynet/extras/daemon/comment_client.py

```diff
@@ -1,31 +1,51 @@
 import logging
+
 import aiohttp
+import hashlib
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.asymmetric import utils
+from cryptography.hazmat.primitives.asymmetric.rsa import generate_private_key
 
 log = logging.getLogger(__name__)
 
 
+def sign_comment(**kwargs):
+    private_key = generate_private_key(
+        public_exponent=65537,
+        key_size=4096,
+        backend=default_backend()
+    )
+    chosen_hash = hashes.SHA256()
+    hasher = hashes.Hash(chosen_hash, default_backend())
+    value_to_hash = b':'.join(bytes(v, 'utf-8') for v in kwargs.values() if type(v) is str)
+    hasher.update(value_to_hash)
+    digest = hasher.finalize()
+    signature = private_key.sign(
+        digest,
+        padding.PSS(
+            mgf=padding.MGF1(hashes.SHA256()),
+            salt_length=padding.PSS.MAX_LENGTH
+        ),
+        utils.Prehashed(chosen_hash)
+    )
+    m = hashlib.sha3_256()
+    m.update(signature)
+    return m.hexdigest()
```

Comment:
I don't understand what's happening here... why is there a private key being generated in a function that's supposed to just sign something?

where is the code for signing a comment with a channel like the stream claims are signed by channels?

Reply:
Placeholder since I wanted to ensure core functionality before worrying about replicating the signing mechanism to mirror that of the API